src/SDK/Resource/ResourceInfo.php

```diff
@@ -122,4 +125,25 @@ public function getSchemaUrl(): ?string
     {
         return $this->schemaUrl;
     }
+
+    private function mergeWith(ResourceInfo $resource): ResourceInfo
+    {
+        $attributes = $this->attributes->toArray() + $resource->getAttributes()->toArray();
+
+        $schemaUrl = null;
+        if (null === $this->schemaUrl) {
+            $schemaUrl = $resource->getSchemaUrl();
+        } elseif (null === $resource->getSchemaUrl()) {
+            $schemaUrl = $this->schemaUrl;
+        } elseif ($this->schemaUrl === $resource->getSchemaUrl()) {
+            $schemaUrl = $this->schemaUrl;
+        } else {
```

Comment:
Please refactor these conditions in a method which simply resolves the schemaURL and remove else/elseif.

Reply:
This piece of code is self-describing and follows [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/441bafa8eda1697c3f0b975250e421e25da496b8/specification/resource/sdk.md#merge). A simplified version could hide an original logic.

Reply:
I wouldn't go for a simplified version, but rather avoiding the "IFs" with return statements.
I know it's to be found like that in the spec, which doesn't really make the spec better in my eyes.
However avoiding "IFs" is my personal issue, so I'd at least ask for it. Feel free to ignore my request. :)